### PR TITLE
Update showdown to v0.5.1 and convert NPM updating to git updating

### PR DESCRIPTION
**Pull request by NathanJang**

### CODE REVIEW

**Comment by NathanJang** [comment]
From `npm info showdown`, 0.5.1 is not there, but it is a release on showdown's GitHub repo.

Actually, the new NPM configuration that points to `dist` instead of `compressed` is only under v1.0.0, which is currently in alpha. Should I hold off on updating to `dist`?

**Comment by PeterDaveHello** [comment]
What about just move on git auto-update instead of npm?
btw, there are 4 files under the dist folder in the git repo.

**Comment by NathanJang** [comment]
If you checkout the 0.5.1 tag that's what's under compressed.

**Comment by PeterDaveHello** [comment]
And seems there should be also a folder - `extensions`

**Comment by NathanJang** [comment]
Whoops, I'll add it and look into git auto updating when I get home.

**Comment by NathanJang** [comment]
@PeterDaveHello I added the files and added git autoupdating.

File: ajax/libs/showdown/package.json [hunk 1]
```diff
@@ -31,15 +31,17 @@
...
-        "*.map"
-      ]
-    }
-  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/showdownjs/showdown.git",
+    "basePath": "dist",
+    "files": [
+      "showdown.js",
```
Comment (PeterDaveHello) [inline ajax/libs/showdown/package.json]: Please set to `**/*` in include them all.

File: ajax/libs/showdown/package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "showdown",
   "filename": "showdown.min.js",
   "npmName": "showdown",
```
Comment (PeterDaveHello) [inline ajax/libs/showdown/package.json]: Please remove `npmName` field, it's for npm auto-update only

**Comment by PeterDaveHello** [comment]
Last thing, please say you transfer this lib to use git auto-update in commit message, thanks!

**Comment by NathanJang** [comment]
@PeterDaveHello done!

**Comment by PeterDaveHello** [comment]
Thanks!

**Comment by NathanJang** [comment]
By the way, :+1: on the sparse checkout tutorial. It was really helpful.

**Comment by PeterDaveHello** [comment]
Nice, glad you like it.

**Pull request merged**